### PR TITLE
Add SSOTokenImpl serialization test for jcache

### PR DESCRIPTION
**Pull request by ryanesch**
... for https://github.com/OpenLiberty/open-liberty/issues/20568

### CODE REVIEW

File: dev/com.ibm.ws.security.saml.websso.2.0/test/com/ibm/ws/security/saml/sso20/token/Saml20TokenImplSerializationTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        Saml20TokenImpl object = (Saml20TokenImpl) in.readObject();
+
+        in.close();
+
+        //Check issuer
+        assertEquals("The issuer should be: https://localhost:8960/idp/shibboleth", "https://localhost:8960/idp/shibboleth", object.getSAMLIssuerName());
+
+        //Check samlId
+        assertEquals("The samlId should be _5a9aa7a5baba3e58916977ebb2345a1f", "_5a9aa7a5baba3e58916977ebb2345a1f", object.getSamlID());
+
```
Comment (jvanhill) [inline dev/com.ibm.ws.security.saml.websso.2.0/test/com/ibm/ws/security/saml/sso20/token/Saml20TokenImplSerializationTest.java:83]: We should be testing all of the fields that we can test.

**Review by jvanhill — CHANGES_REQUESTED** [review]

File: dev/com.ibm.ws.security.saml.websso.2.0/test/com/ibm/ws/security/saml/sso20/token/Saml20TokenImplSerializationTest.java (hunk 1 above)
Comment (ryanesch) [reply dev/com.ibm.ws.security.saml.websso.2.0/test/com/ibm/ws/security/saml/sso20/token/Saml20TokenImplSerializationTest.java:83]: Updated the test to check the samlString. This is used during deserialization to populate the many other fields, so it should suffice to only check samlString I think.

File: dev/com.ibm.ws.security.saml.websso.2.0/test/com/ibm/ws/security/saml/sso20/token/Saml20TokenImplSerializationTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        in.close();
+
+        //Check serializationVersion via reflection
+        Field field = Saml20TokenImpl.class.getDeclaredField("serialVersionUID");
+        field.setAccessible(true);
+        Object value = field.get(object);
+        assertEquals("The version should be -862850937499495719L", -862850937499495719L, value);
+
+        //Check samlString
```
Comment (jvanhill) [inline dev/com.ibm.ws.security.saml.websso.2.0/test/com/ibm/ws/security/saml/sso20/token/Saml20TokenImplSerializationTest.java:79]: Add comment explaining that you only check the saml string b/c everything else in the object is derived from the string.

File: dev/com.ibm.ws.security.token.ltpa/test/com/ibm/ws/security/token/ltpa/internal/LTPAToken2SerializationTest.java [hunk 3]
```diff
@@ -76,9 +76,20 @@ public void deserializeLTPAToken2_1() throws Exception {
...
+        assertEquals("The version should be 1", 1, object.getVersion());
+
         //Check token
+        //The userData, privateKey, sharedKey, and cipher are used to return the byte array.
         assertEquals("The token byte string should be: " + getByteString(test), getByteString(test), getByteString(object.getBytes()));
 
+        //Validate the token
+        //Validation uses signature and publicKey.
+        assertEquals("The token should be valid.", true, object.isValid());
+
```
Comment (jvanhill) [inline dev/com.ibm.ws.security.token.ltpa/test/com/ibm/ws/security/token/ltpa/internal/LTPAToken2SerializationTest.java:98]: To test:

- getAttributeNames()
- getAttributes()
- getExpiration()

File: dev/com.ibm.ws.security.token/test/com/ibm/ws/security/token/internal/SerializationTest.java [hunk 4]
```diff
@@ -33,7 +33,7 @@
 public class SerializationTest {
```
Comment (jvanhill) [inline dev/com.ibm.ws.security.token/test/com/ibm/ws/security/token/internal/SerializationTest.java]: Rename this to SIngleSignonTokenImplSerializationTest

**Review by jvanhill — CHANGES_REQUESTED** [review]

File: dev/com.ibm.ws.security.saml.websso.2.0/test/com/ibm/ws/security/saml/sso20/token/Saml20TokenImplSerializationTest.java (hunk 2 above)
Comment (ryanesch) [reply dev/com.ibm.ws.security.saml.websso.2.0/test/com/ibm/ws/security/saml/sso20/token/Saml20TokenImplSerializationTest.java:79]: It already exists in the method comment, but I can add it here too.

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by jvanhill** [comment]
This work was completed in #22972.

**Pull request closed without merging**